nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -294,7 +294,7 @@ def get_inference_config(self):
     def model_provider_func(self, pre_process, post_process):
         """Model depends on pipeline paralellism."""
         model = GPTModel(
-            vocab_size=self.padded_vocab_size,
+            vocab_size=self.cfg.get('override_vocab_size', self.padded_vocab_size),
```

Comment:
This is needed for converting models from one platform (Mosaic, HF, etc) into Megatron, because you need to instantiate the model with your Embedding layer for the exact dimensionality you require in order to import the weights, otherwise you get a mismatch or you need to do complex weight surgery to cut out the weights you need and pad with zeros. It was easier to add a special override flag to just force the model to use the vocab_size necessary to make downstream conversion work. If the `override_vocab_size` is None, which it will be 99.8% of the time, it uses padded_vocab_size same as before, so it felt like a really harmless bypass to make model conversion much easier for all future conversion work.

Reply:
So, for MPT-7B, the override_vocab_size is not equal to the padded vocab size computed based on the tokenizer vocab size plus divisibility by 128 upscaling?

I have concerns about this for two reasons 1) This is a "dangling" property with no reference in the yaml file so the only way for people to use this is to override from CLI via something like `+model.override_vocab_size=32000` 2) There are divisibility constraints on the vocab size based on the tensor parallel size and mod 128 divisibility for tensor cores, this sidesteps both of them.

Reply:
@MaximumEntropy for the final call, but @trias702, if it's for external only - how about renaming it to make it clear or adding a comment?